### PR TITLE
Check for SDF files containing sibling elements of the same type with the same name (Gazebo9)

### PR DESCRIPTION
**Pull request by j-rivero**
~~**Prerequisite:** this requires https://github.com/osrf/sdformat/pull/580 . In Draft until then.~~
The PR adds a check for SDF files containing sibling elements with the same type. This is part of the [SDF specs ](http://sdformat.org/tutorials?tut=pose_frame_semantics&ver=1.5&cat=specification&#element-naming-rules-in-sdf-1-4) but was not enforced or check until now in Gazebo.

Since this PR will change the output behavior in the series of Gazebo9 releases I'm adding a check for an environment variable `GAZEBO9_BACKWARDS_COMPAT_WARNINGS_ERRORS` that users can set to avoid this new behavior if users want to preserve current behavior.

### CODE REVIEW

**Comment by j-rivero** [comment]
@osrf-jenkins run tests

**Comment by j-rivero** [comment]
brew is blocked by https://github.com/osrf/homebrew-simulation/pull/1487

**Comment by j-rivero** [comment]
abichecker is broken and needs https://github.com/ignition-tooling/release-tools/pull/470 . A valid build could be [Build Status](https://build.osrfoundation.org/job/gazebo-abichecker-any_to_any-ubuntu_auto-amd64/884/).

**Comment by j-rivero** [comment]
Windows CI: [Build Status](https://build.osrfoundation.org/job/gazebo-ci-pr_any-windows7-amd64/3243/)

**Comment by j-rivero** [comment]
Hanging CI run should be the same than: [Build Status](https://build.osrfoundation.org/job/gazebo-ci-pr_any-ubuntu_auto-amd64-gpu-nvidia/668/)

File: gazebo/Server.cc
Comment (jennuine) [inline gazebo/Server.cc:73]: Does this need the same checks as in #3017 ?
https://github.com/osrf/gazebo/blob/0219c47574def474db97355b3b0517aaabf37de5/gazebo/Server.cc#L87-L95

File: test/integration/sdf_errors.cc
Comment (jennuine) [inline test/integration/sdf_errors.cc]: This file/test differs from #3016 , should they be the same?

File: gazebo/Server.cc
Comment (j-rivero) [reply gazebo/Server.cc:73]: gazebo9 only support SDF spec until 1.6 so we don't that need logic in this case.

File: test/integration/sdf_errors.cc
Comment (j-rivero) [reply test/integration/sdf_errors.cc]: It will make things easier for merging forward, good catch: b7cc563

File: test/integration/sdf_errors.cc
Comment (jennuine) [inline test/integration/sdf_errors.cc]: minor nit: in #3017 the comment is after `Load` (may give future merge forward conflicts)
```suggestion
  Load("worlds/test_sdf16_err_sibling_different_type.world");
  // 1.6 SDF does not enforce different names for different types
```

**Review by jennuine — APPROVED** [review]
LGTM. Not sure what's going on with CI though...

**Comment by j-rivero** [comment]
> LGTM. Not sure what's going on with CI though...

several fires :fire: are ongoing in the buildfarm making things hard. I'm fighting with them. Thanks Jenn

**Comment by j-rivero** [comment]
CI is now happy.

**Pull request merged**
